src/pendingactions.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module core/pendingactions
+ */
+
+/* global window */
+
+import Plugin from './plugin';
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+import DOMEmitterMixin from '@ckeditor/ckeditor5-utils/src/dom/emittermixin';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+
+/**
+ * List of editor pending actions.
+ *
+ * Any asynchronous action that should be finished before the editor destroy (like file upload) should be registered
```

Comment:
> should be finished before the editor destroy

This plugin has nothing to do with editing destroy. It is to handle long-lasting actions. To synchronise plugins which execute such action (i.e. file upload) and the editor integration. It gives a developer, who integrates the editor, an easy way to check if there are any pending action whenever such information is needed. All plugins, which register pending action provides also a message what action is ongoing which can be displayed to a user and let him decide if he wants to interrupt the action or wait. This plugin is not able to interrupt actions or manage that in any other way.